operator/main.go

```diff
@@ -161,6 +181,85 @@ func runOperator(cmd *cobra.Command) {
 	}
 	close(k8sInitDone)
 
+	k8sversion.Update(k8s.Client(), option.Config)
+	capabilities := k8sversion.Capabilities()
+
+	if !capabilities.MinimalVersionMet {
+		log.Fatalf("Minimal kubernetes version not met: %s < %s",
+			k8sversion.Version(), k8sversion.MinimalVersionConstraint)
+	}
+
+	// If the K8S version has support for coordination.k8s.io/v1 use LeasesResourceLock
+	// for resourcelock config else fallback to non HA mode for Operator.
+	// We only support operator in HA mode for Kubernetes Versions having support for
+	// LeasesResourceLock
+	// LeasesResourceLock type is preferred for resourcelocks because edits to Leases are less common
+	// and fewer objects in the cluster watch "all Leases".
```

Comment:
```suggestion
	// If the K8s version has support for coordination.k8s.io/v1 use LeasesResourceLock
	// for resourcelock config else fallback to non HA mode for Operator.
	// We only support Operator in HA mode for Kubernetes Versions having support for
	// LeasesResourceLock.
	// LeasesResourceLock type is preferred for resourcelocks because edits to Leases are less common
	// and fewer objects in the cluster watch "all Leases".
```

I think the majority of this comment can be folded into the main documentation for the capability itself. We can keep a short explanation of why we're doing this check and then something like "See docs on `capabilities.LeasesResourceLock` for more context."

Reply:
Makes sense, I will update the documentation.